### PR TITLE
Add nftables support.

### PR DESCRIPTION
**Pull request by ipuustin**
Move Refkit firewall support from iptables to nftables. The benefit of the new framework would be file-based configuration of firewall rules and the ability to assign interfaces to zones. Also, in the framework the firewall rules are executed as a single ruleset, meaning ports might be opened already before the port user is started up. The firewall transactions are now completely atomic.

### CODE REVIEW

**Comment by ipuustin** [comment]
Hmm, looks like `refkit-image-common` doesn't have `python3`...

**Comment by ipuustin** [comment]
retest this please

**Comment by ipuustin** [comment]
retest this please

**Comment by ipuustin** [comment]
Systemd changes are tracked here: https://github.com/systemd/systemd/pull/5839

**Comment by ipuustin** [comment]
retest this please

**Comment by ipuustin** [comment]
Rebased.

**Comment by ipuustin** [comment]
retest this please

**Comment by ipuustin** [comment]
retest this please

**Comment by ipuustin** [comment]
Rebased, should be now Yocto 2.0 compatible.

File: meta-refkit-core/recipes-connectivity/linux-intel/linux-intel_%.bbappend [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+SRC_URI_append = " file://nftables.cfg"
```
Comment (mythi) [inline meta-refkit-core/recipes-connectivity/linux-intel/linux-intel_%.bbappend]: yocto-kernel-cache has nf_tables KERNEL_FEATURE. Have you submitted these in yocto-kernel-cache yet? Or would the existing nf_tables be enough?

Comment (ipuustin) [inline meta-refkit-core/recipes-connectivity/linux-intel/linux-intel_%.bbappend]: It's not enough... I'll submit a patch.

**Comment by ipuustin** [comment]
retest this please

**Comment by ipuustin** [comment]
test this please

**Comment by ipuustin** [comment]
test this please

**Comment by ipuustin** [comment]
test this please

**Comment by ipuustin** [comment]
test this please

**Comment by Arthil** [comment]
There was a problem in CI, BBB froze for some reason. Should be ok now. 
Test this please.

**Comment by ipuustin** [comment]
retest this please

File: meta-refkit-core/recipes-security/nftables-settings-default/files/firewall-update.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+output_data = data.format(local_interfaces=local_ifs, lan_interfaces=lan_ifs,
+        wan_interfaces=wan_ifs, dmz_interfaces=dmz_ifs,
+        vpn_interfaces=vpn_ifs, all_interfaces=all_ifs)
+
+# Do not write the ruleset file if it already exists and there is no change.
+# This prevents unneccessary firewall setup changes.
+
+current_data = None
+
+if (os.path.exists(zonesRulesetPath)):
```
Comment (rojkov) [inline meta-refkit-core/recipes-security/nftables-settings-default/files/firewall-update.py]: redundant braces.
I guess it wouldn't harm to check this script with pylint.

**Comment by ipuustin** [comment]
@rojkov : Thanks for the review comments. I did the changes you requested.

**Comment by ipuustin** [comment]
Rebased.

**Comment by mythi** [comment]
test this please

**Comment by mythi** [comment]
@ipuustin looks like this is now failing in networking tests

**Comment by ipuustin** [comment]
@mythi Yes. Perhaps the rebase went somehow wrong. I'll investigate now.

**Comment by ipuustin** [comment]
Might have caught it. Related to iotivity test refactoring. Retest this please.

**Comment by ipuustin** [comment]
test this please

**Comment by ipuustin** [comment]
The test error doesn't appear to be related. Retest this please anyway.

**Comment by ipuustin** [comment]
retest this please

**Pull request merged**